Release/src/http/client/http_client_asio.cpp

```diff
@@ -252,7 +255,10 @@ class asio_connection_pool
     m_io_service(io_service),
     m_timeout_secs(static_cast<int>(idle_timeout.count())),
     m_start_with_ssl(start_with_ssl),
-    m_ssl_context_callback(ssl_context_callback)
+    m_ssl_context_callback(ssl_context_callback),
+    m_pool_timeout_secs(60), // Clean this connection pool 60 secs after the last asio_client release it.
+    m_pool_timer(io_service),
+    m_use_count(0)
```

Comment:
But it's possible that at that point some http_client objects still hold the pool, although all individual connections have closed.  In this case we cannot clean up the entry.  I agree that manual reference counting is error-prone, so in the new change I removed the manual reference counting, and rely on shared_ptr's use_count.